src/acrn/acrn_driver.c

```diff
@@ -2556,6 +2556,50 @@ acrnInitPlatform(acrnPlatformInfoPtr pi, virNodeInfoPtr nodeInfo,
     return ret;
 }
 
+static int
+acrnPersistentDomainInit(virDomainObjPtr dom, void *opaque)
+{
+    int ret = -1;
+    acrnConnectPtr privconn = opaque;
+
+    unsigned char hvUUID[VIR_UUID_BUFLEN];
+    struct acrnVmList *vmList;
+    acrnDomainObjPrivatePtr priv;
+    virObjectEventPtr event = NULL;
+
+    vmList = acrnVmListNew();
+    if (!vmList)
+        goto cleanup;
+
+    acrnDriverLock(privconn);
+
+    if (acrnGetPlatform(&privconn->pi, vmList) < 0)
+        goto cleanup_unlock;
+
+    if (acrnAllocateVm(privconn->domains, dom->def, &privconn->pi, vmList,
+        false, hvUUID) < 0)
+        goto cleanup_unlock;
+
+    priv = dom->privateData;
+    uuid_copy(priv->hvUUID, hvUUID);
+
+    event = virDomainEventLifecycleNewFromObj(dom,
+                                              VIR_DOMAIN_EVENT_DEFINED,
```

Comment:
Let's make this a bit more robust here. We can goto cleanup_unlock if !event.

Reply:
Again, checking the event pointer and bailing out with an error seem correct. Will be in the next patch ...